upup/pkg/fi/cloudup/awstasks/iamrolepolicy.go

```diff
@@ -43,12 +43,49 @@ type IAMRolePolicy struct {
 
 	// The PolicyDocument to create as an inline policy.
 	// If the PolicyDocument is empty, the policy will be removed.
-	PolicyDocument fi.Resource
+	PolicyDocument  fi.Resource
+	PolicyOverrides *[]string
+	Managed         bool
 }
 
 func (e *IAMRolePolicy) Find(c *fi.Context) (*IAMRolePolicy, error) {
+	var actual IAMRolePolicy
+
 	cloud := c.Cloud.(awsup.AWSCloud)
 
+	// Handle policy overrides
+	if e.PolicyOverrides != nil {
+		request := &iam.ListAttachedRolePoliciesInput{
+			RoleName: e.Role.Name,
+		}
+
+		response, err := cloud.IAM().ListAttachedRolePolicies(request)
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "NoSuchEntity" {
+				return nil, nil
+			}
+
+			return nil, fmt.Errorf("error getting policies for role: %v", err)
+		}
+
+		policies := make([]string, 0)
```

Comment:
Nit: Normally it's fine just to do `var policies []string` - unless you're making a distinction between the empty list and nil (which is typically a bad idea, though I'm sure I've done it!)